google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1113,4 +1114,289 @@ public Bucket setDefaultKmsKey(String bucketName, String kmsKeyName) throws Stor
+
+    // The name of a blob, e.g. "my-blob"
+    // String blobName = "my-blob";
+
+    // Select all fields
+    // Fields can be selected individually e.g. Storage.BlobField.CACHE_CONTROL
+    Blob blob = storage.get(bucketName, blobName, BlobGetOption.fields(Storage.BlobField.values()));
+
+    // Print blob metadata
+    System.out.println("Bucket: " + blob.getBucket());
+    System.out.println("CacheControl: " + blob.getCacheControl());
+    System.out.println("ComponentCount: " + blob.getComponentCount());
+    System.out.println("ContentDisposition: " + blob.getContentDisposition());
+    System.out.println("ContentEncoding: " + blob.getContentEncoding());
+    System.out.println("ContentLanguage: " + blob.getContentLanguage());
+    System.out.println("ContentType: " + blob.getContentType());
+    System.out.println("Crc32c: " + blob.getCrc32c());
+    System.out.println("ETag: " + blob.getEtag());
+    System.out.println("Generation: " + blob.getGeneration());
+    System.out.println("Id: " + blob.getBlobId());
+    System.out.println("KmsKeyName: " + blob.getKmsKeyName());
+    System.out.println("Md5Hash: " + blob.getMd5());
+    System.out.println("MediaLink: " + blob.getMediaLink());
+    System.out.println("Metageneration: " + blob.getMetageneration());
+    System.out.println("Name: " + blob.getName());
+    System.out.println("Size: " + blob.getSize());
+    System.out.println("StorageClass: " + blob.getStorageClass());
+    System.out.println("TimeCreated: " + new Date(blob.getCreateTime()));
+    System.out.println("Last Metadata Update: " + new Date(blob.getUpdateTime()));
+    System.out.println(
+        "temporaryHold: "
+            + (blob.getTemporaryHold() != null && blob.getTemporaryHold()
+                ? "Enabled"
+                : "Disabled"));
+    System.out.println(
+        "eventBasedHold: "
+            + (blob.getTemporaryHold() != null && blob.getTemporaryHold()
+                ? "Enabled"
+                : "Disabled"));
+    if (blob.getRetentionExpirationTime() != null) {
+      System.out.println("retentionExpirationTime: " + new Date(blob.getRetentionExpirationTime()));
+    }
+    if (blob.getMetadata() != null) {
+      System.out.println("User metadata:");
+      for (Map.Entry<String, String> userMetadata : blob.getMetadata().entrySet()) {
+        System.out.println(userMetadata.getKey() + " " + userMetadata.getValue());
+      }
+    }
+    // [END storage_get_metadata]
+  }
+
+  /** Example of setting a retention policy on a bucket */
+  public Bucket setRetentionPolicy(String bucketName, Long retentionPeriod)
+      throws StorageException {
+    // [START storage_set_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    // The retetion period for objects in bucket
+    // 1 hour in seconds
```

Comment:
nit: Move this comment to the next line (`3600L; // 1 hour`) (Otherwise it feels a little unclear what it's describing) 

Also s/retetion/retention

Reply:
ack